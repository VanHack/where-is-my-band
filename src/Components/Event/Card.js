import styled from "styled-components";

const Card = styled.div`
  list-style-type: none;
  background: #7b1fa2;
  color: white;
  padding: 15px;
  border-radius: 0px 3px 3px 0px;
  width: 100%;
  margin: 15px;
  transition: all 0.2s ease-in-out;

  &:hover {
    transform: scale(1.04);
    cursor: pointer;
  }

  @media (min-width: 1200px) {
    flex-basis: 40%;
    flex-grow: 1;
  }
`;

export default Card;
