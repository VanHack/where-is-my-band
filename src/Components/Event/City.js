import styled from "styled-components";

const City = styled.div`
  font-size: 20px;
`;

export default City;
