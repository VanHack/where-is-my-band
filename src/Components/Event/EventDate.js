import styled from "styled-components";

const EventDate = styled.div`
  display: inline-block;
  margin-left: auto;
  font-size: 20px;
`;

export default EventDate;
