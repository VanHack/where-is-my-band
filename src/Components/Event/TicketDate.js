import styled from "styled-components";

const TicketDate = styled.div`
  font-size: 20px;
`;
export default TicketDate;
