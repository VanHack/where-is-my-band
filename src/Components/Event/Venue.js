import styled from "styled-components";

const Venue = styled.h1`
  font-size: 30px;
  color: white;
  font-weight: bold;
  margin: 0 0 15px;
`;

export default Venue;
