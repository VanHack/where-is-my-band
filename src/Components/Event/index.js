import React from "react";

import Card from "./Card";
import Venue from "./Venue";
import City from "./City";
import EventDate from "./EventDate";
import TicketDate from "./TicketDate";

const Event = ({
  data: { venue, datetime: date, on_sale_datetime: onSale }
}) => (
  <Card>
    <Venue>{venue.name}</Venue>
    <City>{`${venue.city}, ${venue.country}`}</City>
    <EventDate>Date: {new Date(date).toLocaleDateString()}</EventDate>
    <TicketDate>
      Tickets available:
      {onSale ? new Date(onSale).toLocaleDateString() : " TBA"}
    </TicketDate>
  </Card>
);

export default Event;
