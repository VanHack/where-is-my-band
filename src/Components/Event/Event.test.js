import React from 'react';
import Event from './index';
import renderer from 'react-test-renderer';

test('renders an event completely', () => {
  const myEvent = {
    venue: {
      name: 'MyVenue',
      datetime: new Date(),
      on_sale_datetime: new Date()
    }
  }

  const component = renderer.create(
    <Event data={myEvent}/>,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
