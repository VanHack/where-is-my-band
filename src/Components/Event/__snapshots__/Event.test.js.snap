// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an event completely 1`] = `
<div
  className="sc-bdVaJa dWSdXx"
>
  <h1
    className="sc-bwzfXH hYipUd"
  >
    MyVenue
  </h1>
  <div
    className="sc-htpNat fIjutD"
  >
    undefined, undefined
  </div>
  <div
    className="sc-bxivhb llqzqF"
  >
    Date: 
    Invalid Date
  </div>
  <div
    className="sc-ifAKCX fQmBEu"
  >
    Tickets available:
     TBA
  </div>
</div>
`;
